### PR TITLE
open-isns: 0.99 -> 0.100

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
https://github.com/open-iscsi/open-isns/blob/v0.100/ChangeLog

###### Things done
* Update + add upstream patches
* Update Github repo source and homepage
* add maintainer
* Fix attribute to match official name

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change.
 Tested with out of tree test: https://github.com/markuskowa/nix-system/blob/master/tests/isns.nix
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/open-isns/default.nix [hunk 1]
```diff
@@ -20,10 +33,11 @@ stdenv.mkDerivation rec {
   installFlags = [ "etcdir=$(out)/etc" "vardir=$(out)/var/lib/isns" ];
   installTargets = [ "install" "install_hdrs" "install_lib" ];
 
-  meta = {
+  meta = with lib; {
     description = "iSNS server and client for Linux";
-    license = lib.licenses.lgpl21;
-    homepage = "https://github.com/gonzoleeman/open-isns";
-    platforms = lib.platforms.linux;
+    license = licenses.lgpl21;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/open-isns/default.nix:38]: lgpl21Only or lgpl21Plus?

Comment (markuskowa) [reply pkgs/os-specific/linux/open-isns/default.nix]: Fixed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111009` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>open-isns</li>
<li>openiscsi</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

openiscsi:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '/sbin' doesn't match anything in file '/nix/store/j7nr2xfsay6s8kzf7gkdbi61a8vg1wqs-open-iscsi-2.1.3/lib/systemd/system/iscsid.socket'
substituteStream(): WARNING: pattern '/sbin' doesn't match anything in file '/nix/store/j7nr2xfsay6s8kzf7gkdbi61a8vg1wqs-open-iscsi-2.1.3/lib/systemd/system/iscsiuio.socket'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by markuskowa** [comment]
I'll fix the `offending substituteInPlace` in open-iscsi in another PR. This is out of scope here.

**Review by zaninime — APPROVED** [review]

**Pull request merged**